aten/src/ATen/native/TensorConversions.cpp

```diff
@@ -1619,13 +1662,13 @@ Tensor sparse_compressed_to_sparse_bsr(const Tensor& self, IntArrayRef blocksize
     return sparse_compressed_clone(self, blocksize, "to_sparse_bsr");
   }
   if (self.layout() == kSparseCsr) {
-    TORCH_CHECK(self.dim() == 2,
-        "to_sparse_bsr(): conversion from Csr to Bsr is only possible for 2d inputs, ",
-        "but got input of dimension ", self.dim(), " instead.");
+    TORCH_CHECK(self.dim() == 2 + self.dense_dim(),
+                "to_sparse_bsr: conversion from Csr to Bsr for batched inputs is not implemented.");
+
     Tensor self_values = self.values();
     Tensor self_crow_indices = self.crow_indices();
     Tensor self_col_indices = self.col_indices();
-    Tensor cpu_result = _csr_to_block_csr_cpu(
+    Tensor cpu_result = _compressed_to_block_compressed_cpu<true>(
```

Comment:
I wonder if we could have here:
```suggestion
    return _compressed_to_block_compressed_cpu<true>(self.cpu(), blocksize()).to(self.device());
```
or similar?

Reply:
So, the conversion only takes place on the CPU? Hmm...

Reply:
Changed.  As a reminder: this PR is just refactoring existing CSR->BSR conversion (that, indeed, is slow and taking place on the CPU) so that CSC->BSC, that was not there before, is supported in the same way too.